Emby.Server.Implementations/Data/SqliteUserRepository.cs

```diff
@@ -1,85 +1,86 @@
+using System.Collections.Generic;
+using System.IO;
+using MediaBrowser.Controller;
+using MediaBrowser.Controller.Entities;
+using MediaBrowser.Controller.Persistence;
+using MediaBrowser.Model.Serialization;
+using Microsoft.Extensions.Logging;
+using SQLitePCL.pretty;
+
+namespace Emby.Server.Implementations.Data
+{
+    /// <summary>
+    /// Class SQLiteUserRepository
+    /// </summary>
+    public class SqliteUserRepository : BaseSqliteRepository, IUserRepository
+    {
+        private readonly IJsonSerializer _jsonSerializer;
+
+        public SqliteUserRepository(
+            ILoggerFactory loggerFactory,
+            IServerApplicationPaths appPaths,
+            IJsonSerializer jsonSerializer)
+            : base(loggerFactory.CreateLogger(nameof(SqliteUserRepository)))
+        {
+            _jsonSerializer = jsonSerializer;
+
+            DbFilePath = Path.Combine(appPaths.DataPath, "users.db");
+        }
+
+        /// <summary>
+        /// Gets the name of the repository
+        /// </summary>
+        /// <value>The name.</value>
+        public string Name => "SQLite";
+
+        /// <summary>
+        /// Opens the connection to the database
+        /// </summary>
+        /// <returns>Task.</returns>
+        public void Initialize()
+        {
+            using (var connection = CreateConnection())
+            {
+                RunDefaultInitialization(connection);
+
+                var localUsersTableExists = TableExists(connection, "LocalUsersv2");
+
+                connection.RunQueries(new[] {
+                    "create table if not exists LocalUsersv2 (Id INTEGER PRIMARY KEY, guid GUID NOT NULL, data BLOB NOT NULL)",
+                    "drop index if exists idx_users"
+                });
+
+                if (!localUsersTableExists && TableExists(connection, "Users"))
+                {
+                    TryMigrateToLocalUsersTable(connection);
+                }
+                RemoveEmptyPasswordHashes();
+            }
+        }
+
+        private void TryMigrateToLocalUsersTable(ManagedConnection connection)
+        {
+            try
+            {
+                connection.RunQueries(new[]
+                {
+                    "INSERT INTO LocalUsersv2 (guid, data) SELECT guid,data from users"
+                });
+            }
+            catch (Exception ex)
+            {
+                Logger.LogError(ex, "Error migrating users database");
+            }
+        }
+
         private void RemoveEmptyPasswordHashes()
         {
             foreach (var user in RetrieveAllUsers())
             {
                 // If the user password is the sha1 hash of the empty string, remove it
-                if (!string.Equals(user.Password, "DA39A3EE5E6B4B0D3255BFEF95601890AFD80709") || !string.Equals(user.Password, "$SHA1$DA39A3EE5E6B4B0D3255BFEF95601890AFD80709"))
+                if (!string.Equals(user.Password, "DA39A3EE5E6B4B0D3255BFEF95601890AFD80709", StringComparison.Ordinal)

```

Comment:
what about _not_ hardcoding a hash of an empty string but compute it as a static class member?

Reply:
I don't have a ICryptoPrivuder in this class to do hashing with. This isn't the end of the world, but I would rather not need introduce one if it could be avoided.

Reply:
It's also not something we want to compute. It's a specific value in the database that needs to be migrated to a different format. If the underlying computation changes, the migration fails. Just don't think of it as a hash ;)

Reply:
Okay, let's stop here. No big deal :)